c10/util/thread_name.cpp

```diff
@@ -1,6 +1,7 @@
 #include <c10/util/thread_name.h>
 
 #include <algorithm>
+#include <array>
```

Comment:
i should probably move this to inside the `#ifdef C10_HAS_PTHREAD_SETNAME_NP` block.

and i also wonder, is the reason CI did not catch this issue becuase CI is not testing this code?  Like, are we not using GLIBC 2.12 anywhere on CI?

Reply:
Let's see if we ever build using glibc-2.12 or newer: https://github.com/pytorch/pytorch/pull/128666

But this is a pretty common problem for C++ projects, as various implementations of libc++ (gnu vs clang vs VC++) transitively include different headers (I.e. I suspect `algorithm` should include `array` on most of the platforms)